kiwi/builder/disk.py

```diff
@@ -718,6 +640,121 @@ def create_install_media(self, result_instance: Result) -> Result:
 
         return result_instance
 
+    def _build_main_system(
+        self,
+        device_map: Dict,
+        disk: Disk,
+        system: Any,
```

Comment:
Do we really not know the concrete type of `system` here?

Reply:
no it can be FileSystem or VolumeManager. Dan really to make this absolutely nice a bigger refactoring is required which I'm not starting together with the context manager switch, sorry

Reply:
ok that I could fix with a cast()

Reply:
If it's either of the two classes, how about this then:
```suggestion
        system: Union[FileSystem, VolumeManager],
```

Reply:
I'll try

Reply:
done